sdk/core/Azure.Core/src/ETag.cs

```diff
@@ -12,14 +13,32 @@ namespace Azure
     {
         private const char QuoteCharacter = '"';
         private const string QuoteString = "\"";
-
+        private const string WeakETagPrefix = "W/\"";
+        private const string DefaultFormat = "G";
+        private const string HeaderFormat = "H";
         private readonly string _value;
+        private readonly bool _preserveRawValue;
 
         /// <summary>
         /// Creates a new instance of <see cref="ETag"/>.
         /// </summary>
         /// <param name="etag">The string value of the ETag.</param>
-        public ETag(string etag) => _value = etag;
+        public ETag(string etag)
+        {
+            _value = etag;
+            _preserveRawValue = true;
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="ETag"/>.
+        /// </summary>
+        /// <param name="etag">The string value of the ETag.</param>
+        /// <param name="preserveValue">Indicates whether the value of the etag should be preserved as is rather that formatted with wrapping quotes.</param>
+        private ETag(string etag, bool preserveValue)
```

Comment:
I don't understand this ctor. Formatting is controlled by the format parameter to ToString. There should not be a boolean here.

Reply:
This is to ensure we don't introduce a breaking change for the existing `ToString()` method and also store weak ETags internally in the raw format.

Reply:
What would be the breaking change if we don't have this field, i.e. what existing code would work differently? the existing ToString simply gives back what was passed to the ctor.

Reply:
But the behavior is inconsistent between `Parse` and the ctor.

Reply:
What does the existing (shipped ctor) do when it received a weak etag? I thought it was throwing. If yes, then we would just stop throwing and execute the same code as Parse calls, i.e. normalize to raw etag.

Reply:
No, only `Parse` throws for any reason.